tests/test_harmony_service.py

```diff
@@ -0,0 +1,131 @@
+import logging
+from datetime import datetime
+from pathlib import Path
+from unittest.mock import patch
+
+import harmony_service_lib
+import pystac
+import pytest
+
+from opera_rtc_s1_browse import harmony_service
+
+
+def mock_download(**kwargs) -> str:
+    assert set(kwargs.keys()) == {'url', 'destination_dir', 'logger', 'access_token'}
+
+    assert kwargs['url'] in ('url/to/mock_VV.tif', 'url/to/mock_VH.tif')
+    assert isinstance(kwargs['destination_dir'], str)
+    assert isinstance(kwargs['logger'], logging.Logger)
+    assert kwargs['access_token'] == 'mock-access-token'
+
+    if kwargs['url'] == 'url/to/mock_VV.tif':
+        return 'mock_VV.tif'
+
+    return 'mock_VH.tif'
+
+
+def mock_create_browse_image(**kwargs) -> Path:
+    assert set(kwargs.keys()) == {'co_pol_path', 'cross_pol_path', 'working_dir'}
+
+    assert kwargs['co_pol_path'] == Path('mock_VV.tif')
+    assert kwargs['cross_pol_path'] == Path('mock_VH.tif')
+    assert isinstance(kwargs['working_dir'], Path)
+
+    return Path('path') / 'to' / 'mock_rgb.tif'
+
+
+def mock_stage(**kwargs) -> str:
+    assert set(kwargs.keys()) == {'local_filename', 'remote_filename', 'mime', 'location', 'logger'}
+
+    assert kwargs['local_filename'] == 'path/to/mock_rgb.tif'
+    assert kwargs['remote_filename'] == 'mock_rgb.tif'
+    assert kwargs['mime'] == 'image/tiff'
+    assert kwargs['location'] == 'mock-staging-location'
+    assert isinstance(kwargs['logger'], logging.Logger)
+
+    return 'mock-staged-url'
+
+
+def test_process_item():
+    adapter = harmony_service.HarmonyAdapter(
+        harmony_service_lib.message.Message(
+            {
+                'accessToken': 'mock-access-token',
+                'stagingLocation': 'mock-staging-location',
+            }
+        )
+    )
+    item = pystac.Item(
+        id='mock-pystac-item',
+        geometry=None,
+        bbox=None,
+        datetime=datetime(2024, 1, 1),
+        properties={},
+        assets={
+            'data': pystac.Asset(href='url/to/mock_VH.tif'),
+            'data1': pystac.Asset(href='url/to/mock_VV.tif'),
+        },
+    )
+    expected_result = pystac.Item(
+        id='mock-pystac-item',
+        geometry=None,
+        bbox=None,
+        datetime=datetime(2024, 1, 1),
+        properties={},
+        assets={
+            'rgb_browse': pystac.Asset(
+                href='mock-staged-url', title='mock_rgb.tif', media_type='image/tiff', roles=['visual']
+            )
+        },
+    )
+    with patch('harmony_service_lib.util.download', mock_download), \
```

Comment:
I was expecting MagicMocks here, was there anything in particular that made concrete functions more attractive?

Reply:
I don't know of a way to have a `MagicMock` return two different values depending on the function call arguments, which is required for `mock_download` to return either the VV or VH filename. And then I decided to use concrete functions for the other two mocks for the sake of consistency.

Reply:
It's not quick to do that conditionally, but you can have it return different outputs each time it's called, which is sufficient in a controlled test environment. return VV the first time, VH the second time, then assert after the fact that the mock was called for VV the first time and VH the second time.

Example: https://github.com/ASFHyP3/hyp3-event-monitoring/blob/develop/tests/test_find_new.py#L212